### PR TITLE
Create user crud

### PR DESCRIPTION
**Pull request by juanwongramirez**
## Summary

- Add UserController
- Add UserIdMismatchException
- Add UserNotFoundException
- Create addBook and removeBook methods in UserController

## Trello Card

https://trello.com/c/FTFnTD0m

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    @PutMapping("/{id}")
+    public User updateUser(@RequestBody User user, @PathVariable Long id) {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException();
+        }
+        userRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
+                        String.format(Constants.FORMAT_MESSSAGE_BOOK_NOT_FOUND,id)));
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/controllers/UserController.java]: same

File: src/main/java/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
+                        String.format(Constants.FORMAT_MESSSAGE_BOOK_NOT_FOUND,id)));
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/controllers/UserController.java]: replace by `FORMAT_MESSSAGE_USER_NOT_FOUND`

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/controllers/UserController.java (hunk 2 above)
Comment (juanwongramirez) [reply src/main/java/wolox/training/controllers/UserController.java]: The changes has been loaded

File: src/main/java/wolox/training/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/UserController.java]: Remember to use your custom exception here 😅

File: src/main/java/wolox/training/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        userRepository.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public User updateUser(@RequestBody User user, @PathVariable Long id) {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException();
+        }
+        userRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/UserController.java]: Same here!

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Very nice work! Just a couple of small adjustments and it should be good to go 🙂

File: src/main/java/wolox/training/controllers/UserController.java (hunk 4 above)
Comment (juanwongramirez) [reply src/main/java/wolox/training/controllers/UserController.java]: i have added custom exceptions

**Review by jeymarsonwolox — APPROVED** [review]

File: src/main/java/wolox/training/utils/Constants.java [hunk 5]
```diff
@@ -8,6 +8,9 @@
     public static String MESSSAGE_BOOK_ALREADY_OWNED = "This book already added for this user";
     public static String FORMAT_MESSSAGE_BOOK_ALREADY_OWNED = "The book %s already added for this user";
 
+    public static String FORMAT_MESSSAGE_USER_NOT_FOUND = "The user with id : %d, not found";
+    public static String MESSSAGE_USER_NOT_FOUND = "The user can not be found";
+    public static String MESSSAGE_USER_ID_MISSMATCH = "The id value of user is not the same of the parameter";
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/utils/Constants.java:13]: There's a typo in the word `Message` 😅

**Review by Jimenarosello — APPROVED** [review]
Nice job! just a typo hehe 🚀

**Pull request merged**